server/src/game/player_message.rs

```diff
@@ -0,0 +1,85 @@
+use uuid::Uuid;
+use actix::{Handler, Message, Addr, Context, AsyncContext};
+
+use crate::engine::{WebSocket, websocket_message};
+use crate::game::Player;
+use chrono::{DateTime, Utc};
+
+#[derive(Message)]
+#[rtype(result = "()")]
+pub struct Connected {
+    pub websocket_addr: Addr<WebSocket>,
+    pub session_id: Uuid,
+}
+
+impl Handler<Connected> for Player {
+    type Result = ();
+
+    fn handle(&mut self, msg: Connected, ctx: &mut Context<Self>) -> Self::Result {
+        if let None = self.websocket_addr {
```

Comment:
imho `match` is easier to read and might be better here...the other usages of `if let`, where there's no reasonable default case for match, are ok

Reply:
This whole handler will be redone in the next PR, which focuses on timing out the user. So it will be match, because I will replace the `Option` with better enum.
Sneak peak:
```rust
pub enum PlayerStatus {
    Connecting,
    Connected { websocket_addr: Addr<WebSocket> },
    LostConnection { last_seen: Instant },
}
```

Reply:
yeah, np...I know it's just an initial skeleton :see_no_evil: